fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java

```diff
@@ -0,0 +1,207 @@
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FsApiTest {
+
+    public static final String FODSELSDATO_IDENTIFIER = "fodselsdato0";
+    public static final String PERSONNUMMER_IDENTIFIER = "personnummer0";
+    private static final String PERSON_PATH = "/personer";
+    private static final String STUDENTUNDERVISNING_PATH = "/studentundervisning";
+    private static final String UNDERVISNING_SEMESTER_AR_PATH = "undervisning.semester.ar";
+    private static final String PERSON_PERSONLOPENUMMER_PATH = "person.personlopenummer";
+    private static final String DB_IDENTIFIER = "dbId";
+    private static final String LIMIT_IDENTIFIER = "limit";
+
+    private static final FsCourseData TEST_COURSE_1 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("ABIO6050"), 1, new FsSemester("2022", "VÅR"))));
+    private static final FsCourseData TEST_COURSE_2 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("ABIO6100"), 1, new FsSemester("2022", "VÅR"))));
+    private static final FsCourseData TEST_COURSE_3 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("OPERA6000"), 1, new FsSemester("2022", "VÅR"))));
+    private static final FsCourseData TEST_COURSE_4 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("OPERA6000"), 2, new FsSemester("2022", "HØST"))));
+    private static final FsCourseData TEST_COURSE_5 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("ABIO6100"), 2, new FsSemester("2022", "VÅR"))));
+    private static final FsCourseData TEST_COURSE_6 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("ABIO6100"), 3, new FsSemester("2022", "HØST"))));
+    private static final FsCourseData TEST_COURSE_7 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("OPERAPRA20"), 1, new FsSemester("2022", "VÅR"))));
+    private static final FsCourseData TEST_COURSE_8 = new FsCourseData(
+        new FsCourse(
+            new FsUndervisning(
+                new FsEmne("OPERAPRA3"), 1, new FsSemester("2022", "HØST"))));
```

Comment:
We should look into creating how to generate input test cases and corresponding expected output from in a single function. Easier to keep track of later.

Reply:
fixed